packages/cli/lib/base-generator.js

```diff
@@ -209,6 +209,15 @@ module.exports = class BaseGenerator extends Generator {
         `Package manager '${packageManager}' is not supported. Use ${supported}.`,
       );
     }
+
+    //In case lb4 is run with discover/datasource model, removing unwanted .env
+    if (
```

Comment:
Its better to have this logic in individual generator classes. That will keep this base generator independent of child implementations. This function is overridden there already. Example here - https://github.com/loopbackio/loopback-next/blob/e854c2bc9924331fbb8882a9a87058335cc82939/packages/cli/generators/discover/index.js#L81

Reply:
You are right. Adding it to the relevant generator is better. I'll move the change to the discover generator. 
The confusion or question I had is that why this property .env is being added to the options object. Probably needed at some point but what would be a good place to remove it from the options object?